syncthing.te

```diff
@@ -0,0 +1,45 @@
+policy_module(syncthing, 1.0.0)
+
+# define types
+type syncthing_t;
+type syncthing_exec_t;
+userdom_user_application_domain(syncthing_t, syncthing_exec_t)
+
+type syncthing_config_home_t;
+userdom_user_home_content(syncthing_config_home_t)
+
+# file permissions
+fs_getattr_xattr_fs(syncthing_t)
+
+userdom_manage_user_home_content(syncthing_t)
```

Comment:
On 08/21/2016 11:07 PM, Naftuli Tzvi Kay wrote:
>> @@ -0,0 +1,45 @@
>> +policy_module(syncthing, 1.0.0)
>> +
>> +# define types
>> +type syncthing_t;
>> +type syncthing_exec_t;
>> +userdom_user_application_domain(syncthing_t, syncthing_exec_t)
>> +
>> +type syncthing_config_home_t;
>> +userdom_user_home_content(syncthing_config_home_t)
>> +
>> +# file permissions
>> +fs_getattr_xattr_fs(syncthing_t)
>> +
>> +userdom_manage_user_home_content(syncthing_t)
> 
>> That means that you would also need to add type transition rules so that files and dirs created in ~ will be created with an auto type transition from user_home_dir_t to user_home_t
> 
> How should I accomplish that? There are quite a few different folders/types which could be created by `syncthing_t` on start, could be `audio_home_t`, etc.
> 
> 

Also `audio_home_t` is, i suspect, a Redhat refpolicy fork specific
type. Its a fluke in my opinion, should never have been implemented. I
think they implemented that for MPD so that they could restrict mpd
ability to read ~/Audio.

syncthing, in my opinion, should only be able to sync generic user home
content, everything else should be silenty denied.

-- 
Key fingerprint = 5F4D 3CDB D3F8 3652 FBD8  02D5 3B6C 5F1D 2C7B 6B02
https://sks-keyservers.net/pks/lookup?op=get&search=0x3B6C5F1D2C7B6B02
Dominick Grift